neo/SmartContract/InteropService.NEO.cs

```diff
@@ -242,6 +242,7 @@ private static bool Contract_Create(ApplicationEngine engine)
             };
 
             if (!contract.Manifest.IsValid()) return false;
+            if (contract.Manifest.Hash != contract.ScriptHash) return false;
```

Comment:
this was here

https://github.com/shargon/neo/blob/fcdab615e6b29c5721332d15e481bd9d248f6d47/neo/SmartContract/Manifest/ContractManifest.cs#L176